### PR TITLE
Upgrade to VS 2017

### PR DESCRIPTION
**Pull request by natemcmaster**
Remove project.json and upgrade to MSBuild.

Other notable changes:
Removed unused baseline.json files. There were baseline files for "Microsoft.AspNetCore.Server.Testing" package, even though there was no project by that name. cc @javiercn @Eilon

### CODE REVIEW

**Comment by Tratcher** [comment]
Those baselines should be moved from Microsoft.AspNetCore.Server.Testing to Microsoft.AspNetCore.Server.IntegrationTesting

**Comment by natemcmaster** [comment]
That project isn't an automated test _project_. It's a test _app_. `.notest` prevents korebuild from running `dotnet test` on the project.

**Comment by Eilon** [comment]
Regarding baselines, if the project got renamed, there's no point renaming the baseline, because the whole thing is a breaking change. Is this a package we publish? Should we even be checking for breaking changes in it?

**Comment by natemcmaster** [comment]
I'm not sure the baseline files were ever used. They were added in 911da31476615a5ce4bfe2c9a59cffeddb8c2422, but even then, there was no project named Microsoft.AspNetCore.Server.Testing. See https://github.com/aspnet/Hosting/tree/911da31476615a5ce4bfe2c9a59cffeddb8c2422/src/Microsoft.AspNetCore.Server.Testing

File: src/Microsoft.AspNetCore.Hosting.Server.Abstractions/Microsoft.AspNetCore.Hosting.Server.Abstractions.csproj [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="..\..\build\common.props" />
+
+  <PropertyGroup>
+    <Description>ASP.NET Core hosting server abstractions for web applications.</Description>
+    <TargetFrameworks>net451;netstandard1.3</TargetFrameworks>
+    <NoWarn>$(NoWarn);CS1591</NoWarn>
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/Microsoft.AspNetCore.Hosting.Server.Abstractions.csproj]: Should this go into common.props?

Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/Microsoft.AspNetCore.Hosting.Server.Abstractions.csproj]: I think not. It's nice to keep this a little more visible as a reminder that we are missing XML docs in some API

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/Microsoft.AspNetCore.Hosting.Server.Abstractions.csproj]: If we used the `internal` keyword more I would agree. The fact that we have this in every single project is why I think it should go into the common props files.

Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/Microsoft.AspNetCore.Hosting.Server.Abstractions.csproj]: I still prefer it in the project, right next to the GenerateDocumentationFile flag, and for the sake of consistency about what common.props is really for... but I don't care a whole lot either way.

File: samples/SampleStartups/SampleStartups.csproj [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>net451;netcoreapp1.1</TargetFrameworks>
```
Comment (BrennanConroy) [inline samples/SampleStartups/SampleStartups.csproj]: Need a `RuntimeIdentifier` here?

Comment (natemcmaster) [inline samples/SampleStartups/SampleStartups.csproj]: No. This is a class library. RID is only required for exe projects, and as a workaround for https://github.com/dotnet/sdk/issues/396

File: src/Microsoft.AspNetCore.Hosting.Server.Abstractions/Microsoft.AspNetCore.Hosting.Server.Abstractions.csproj (hunk 1 above)
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/Microsoft.AspNetCore.Hosting.Server.Abstractions.csproj]: ...and I'd guess most devs won't care either.

File: samples/SampleStartups/SampleStartups.csproj (hunk 2 above)
Comment (BrennanConroy) [inline samples/SampleStartups/SampleStartups.csproj]: Looking closer its neither, not supposed to be run, just showing examples of how startups work

File: src/Microsoft.AspNetCore.Hosting.Server.Abstractions/Microsoft.AspNetCore.Hosting.Server.Abstractions.csproj (hunk 1 above)
Comment (Eilon) [inline src/Microsoft.AspNetCore.Hosting.Server.Abstractions/Microsoft.AspNetCore.Hosting.Server.Abstractions.csproj]: I'm with @natemcmaster here, we should leave it per-project. We should ideally do what we did in EF and put explicit docs for `.Internal` types saying they're not supported. But, until then...

**Pull request merged**